src/component/vocabulary/CreateVocabulary.tsx

```diff
@@ -62,36 +67,37 @@ export class CreateVocabulary extends AbstractCreateAsset<CreateVocabularyProps,
     }
 
     public onCreate = () => {
-        const vocabulary: Vocabulary = new Vocabulary({
+        const vocabulary = new Vocabulary({
             label: this.state.label,
             iri: this.state.iri,
             comment: this.state.comment
         });
-        if (this.state.resource == null) {
-            vocabulary.addType(VocabularyUtils.VOCABULARY);
-        } else {
-            vocabulary.addType(VocabularyUtils.DOCUMENT_VOCABULARY);
-            vocabulary.document = this.state.resource as Document;
-        }
-        this.props.onCreate(vocabulary);
+        vocabulary.addType(VocabularyUtils.DOCUMENT_VOCABULARY);
+
+        const files = this.state.files;
+        const fileContents = this.state.fileContents;
+        const document = new Document({
+            label: "Document for " + vocabulary.getLabel(),
```

Comment:
Hardcoded label is IMHO a BAD idea. Most of our deployments are in Czech, this will generate English document labels for them (or Czenglish if the vocabulary title is in Czech). This should be at least localized based on UI language.